enhancements/network/nodeport-range-configuration.md

```diff
@@ -0,0 +1,125 @@
+  In other words customers, when they make non-inclusive changes to the service node-port range are responsible
+  for deleting old services and recreating new services that will then get node ports assigned from the
+  new range.
+- Configuration of multiple service node-port ranges even though supported by the Kubernetes API server will not be
+  allowed. Users are responsible for configuring the range to be large enough for their needs.
+  
+## Proposal
+
+The config observation facility in cluster-kube-apiserver-operator will be used to observe the openshift/api's network 
+configuration object for changes to the service node port range. The config observation controller will then apply the
+change to the `apiServerArguments.service-node-port-range` field. There will only be one entry allowed in the slice.
+
+The following changes are proposed and implemented:
+
+- API changes: [PR #660](https://github.com/openshift/api/pull/660)
+- Library-go changes: [PR #826](https://github.com/openshift/library-go/pull/826)
+- cluster-kube-apiserver-operator changes: [PR #894](https://github.com/openshift/cluster-kube-apiserver-operator/pull/894)
+
+Additionally, to realize the configuration of service node port range, the necessary ports need to be opened on the 
+individual nodes. Traditionally, this has been the responsibility of the Openshift Installer. But since we want this
+configuration to be changed post-install, we need some mechanism to allow for a new set of ports to be opened by _an_
+_entity_ that knows how to tweak the infrastructure.
+
+### Implementation Details/Notes/Constraints
+
+#### Early CI
+
+#### Upstream Kubernetes
+
+This feature is already supported in upstream Kubernetes.
+
+#### Installer
+
+The installer has been traditionally responsible to [open the ports](https://github.com/openshift/installer/blob/master/data/data/aws/vpc/sg-worker.tf#L146)
+belonging to the service node-port range on individual platforms. That functionality will need to be handled via some
+other entity that can also monitor for changes to the service node-port range and open up the necessary ports while the cluster 
+is up and running.
+
+### Risks and Mitigations
+
+#### Completion Risks
+
+The mechanism to handle the configuration and opening of ports on different infrastructure platforms is still TBD. 
+That poses a risk for getting this feature completed in 4.6 timeframe. If we can come to a consensus on how we will
+configure the ports on different infrastructure platforms with minimum re-work either in CNO or some other component,
+we may be able to shrink the timeline for delivering this.
+
+#### Functional Risks
+
+The design for opening up ports on the infrastructure in response to the changes to the `cluster` network configuration object
+needs to be fleshed out. This feature is at risk primarily due to the lack of clarity on that front.
+
```

Comment:
There is a risk of collision with the ephemeral ports, it should not be possible overlapping both ranges

```
sudo sysctl -A | grep ip_local_port_range 
net.ipv4.ip_local_port_range = 32768	60999
```

Reply:
That's a good point specially for IPI. UPI is a little bit more forgiving in that we can expect users to configure the service-node-port-range and ip_local_port_range to not conflict. There is also `ip_local_reserved_ports` that we can ask users to configure for UPI to prevent this conflict. Likewise, for IPI, whatever solution we come up with, we need to propagate that to configure `ip_local_reserved_ports` as well. It's hard to guard against any port range configurations that the users may have on the underlying infrastructure in general though.

Reply:
We can expect nothing from the users :-). Eventually someone **will** manage to overlap ranges. If nothing else, this should be called out in documentation.